plugins/net_plugin/include/eosio/net_plugin/auto_bp_peering.hpp

```diff
@@ -130,6 +229,127 @@ class bp_connection_manager {
              established_client_connection(new_connection) && num_established_clients() > self()->connections.get_max_client_count();
    }
 
+   // thread safe
+   // removes invalid entries from msg
+   bool validate_gossip_bp_peers_message( gossip_bp_peers_message& msg ) {
+      if (msg.peers.empty())
+         return false;
+      if (msg.peers.size() != 1 || !msg.peers[0].server_address.empty()) { // initial case, no server_addresses to validate
+         auto valid_address = [](const std::string& addr) -> bool {
+            const auto& [host, port, type] = net_utils::split_host_port_type(addr);
+            return !host.empty() && !port.empty();
+         };
+         std::map<eosio::name, size_t> producers;
+         const gossip_bp_peers_message::bp_peer* prev = nullptr;
+         for (const auto& peer : msg.peers) {
+            if (peer.producer_name.empty())
+               return false;
+            if (!valid_address(peer.server_address))
+               return false;
```

Comment:
seems that one invalid address will cause us to ignore the whole message. Not what the top comment says:
`// removes invalid entries from msg`.